src/main/java/com/google/graphgeckos/dashboard/storage/DatastoreRepository.java

```diff
@@ -37,20 +37,59 @@
  * The relevant fields for the database, which are described in that class are:
  *    - Kind: "revision"
  *    - Key: commit hash
- * The database indexes after all the BuildInfo fields, besides the {@code builders} list.
- * It is unfeasable to query after the list of builders, since it would not be possible to
- * replicate such a list without querying the database for it first, and the complexity
- * of the indexing and querying would be great, without any apparent value.
  *
  * Useful links:
  * - https://googleapis.dev/java/google-cloud-datastore/latest/index.html
  * - https://googleapis.dev/java/spring-cloud-gcp/1.2.2.RELEASE/index.html
  * - https://cloud.google.com/datastore/docs/concepts/
  * - https://spring.io/projects/spring-cloud-gcp#overview
  */
-@SpringBootApplication
-public class GCDataRepository implements DataRepository {
+@Repository
+public class DatastoreRepository implements DataRepository {
+  /**
+   * This class encapsulates all the parameters for the cleanup subroutine.
+   * Also, provides an easy interface for getting the next timestamp before which
+   * to delete entries. It can be used by multiple GCDataRepository instances.
+   */
+  private class DataRetentionParameters {
+    static int ttlTimeUnit;
+    static int ttl;
+
+    DataRetentionParameters(int ttlTimeUnit, int ttl) {
+      this.ttlTimeUnit = ttlTimeUnit;
+      this.ttl = ttl;
+    }
+
+    /**
+     * This method provides a stateless way to retrieve the timestamp before which to
+     * delete all older entries.
+     *
+     * @return Date object with the appropriate "delete before" timestamp.
+     */
+    Date getEarliestAliveTime() {
+      Calendar calendar = Calendar.getInstance();
+      calendar.setTime(new Date());
+      calendar.add(ttlTimeUnit, ttl);
+      return calendar.getTime();
+    }
+  };
+
+  @Autowired
   private DatastoreTemplate storage;
+  private final ScheduledExecutorService cleanScheduler = Executors.newScheduledThreadPool(1);
```

Comment:
Consider using Spring's `@Scheduled` [functionality](https://www.baeldung.com/spring-scheduled-tasks), more specifically its [cron expressions](https://www.baeldung.com/spring-scheduled-tasks) version for scheduling.
[Cron expressions](https://en.wikipedia.org/wiki/Cron) are a versatile, industry-standard way to specify periodic schedules and Spring's `@Scheduled` makes it easy to hook up (you just need a void method that takes no params to annotate; you don't even need to wrap it in Runnable as you need here).

An additional advantage of a cron-based approach compared to using a ScheduledExecutorService is that when we have to re-deploy or re-start the application, the cron-based approach would not us to perform cause additional cleanup calls (as ScheduledExecutorService does not remember the last time it ran if an app gets restarted or re-deployed).